app/src/main/java/com/arctouch/test/ui/activity/MoviesActivity.kt

```diff
@@ -0,0 +1,151 @@
+package com.arctouch.test.ui.activity
+
+import android.arch.lifecycle.Observer
+import android.arch.lifecycle.ViewModelProviders
+import android.databinding.DataBindingUtil
+import android.os.Bundle
+import android.support.v7.widget.LinearLayoutManager
+import android.view.Menu
+import android.view.MenuItem
+import android.widget.Toast
+import com.arctouch.test.R
+import com.arctouch.test.data.model.Config
+import com.arctouch.test.data.model.Genre
+import com.arctouch.test.data.model.Movie
+import com.arctouch.test.databinding.ActivityMoviesBinding
+import com.arctouch.test.ui.adapter.MoviesAdapter
+import com.arctouch.test.ui.commons.EndlessScrollListener
+import com.arctouch.test.vm.*
+import kotlinx.android.synthetic.main.activity_movies.*
+import javax.inject.Inject
+
+
+class MoviesActivity : ArcTouchActivity() {
+
+    @Inject
+    lateinit var configViewModelFactory: ConfigViewModelFactory
+
+    @Inject
+    lateinit var genresViewModelFactory: GenreViewModelFactory
+
+    @Inject
+    lateinit var moviesViewModelFactory: MoviesViewModelFactory
+
+    private lateinit var configViewModel: ConfigViewModel
+    private lateinit var genreViewModel: GenresViewModel
+    private lateinit var moviesViewModel: MoviesViewModel
+
+    private lateinit var dataBinding: ActivityMoviesBinding
+
+    private var adapter: MoviesAdapter? = null
+
+    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
+        menuInflater.inflate(R.menu.menu_movies, menu)
+        return super.onCreateOptionsMenu(menu)
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        return when (item.itemId) {
+            R.id.action_search -> {
+                navigateToSearch()
+                true
+            }
+            else -> super.onOptionsItemSelected(item)
+        }
+    }
+
+    override fun setUp(savedInstanceState: Bundle?) {
+        super.setUp(savedInstanceState)
+        dataBinding = DataBindingUtil.inflate(layoutInflater, R.layout.activity_movies, null, false)
+        setContentView(dataBinding.root)
+        configViewModel = ViewModelProviders.of(this, configViewModelFactory).get(ConfigViewModel::class.java)
```

Comment:
Why use three view models instead of one?

Reply:
The idea is to separate each view model to make it reusable and control its own live data individually. Whenever we need config or genres they can be used in another activities just as we use in MoviesActivity and SearchMoviesActivity for example.

Reply:
Ok, got it, but usually the `ViewModel` reflects the state of the view. This way, it looks like your view has three states.